services/cd-service/pkg/service/version_test.go

```diff
@@ -166,3 +173,167 @@ func TestVersion(t *testing.T) {
+				},
+			},
+		}
+	}
+	fixtureRelease := func(application string, release uint64) *repository.CreateApplicationVersion {
+		return &repository.CreateApplicationVersion{
+			Application: application,
+			Version:     release,
+			Manifests: map[string]string{
+				"development": rand.String(512),
+				"staging":     rand.String(512),
+			},
+			SourceCommitId: rand.String(64),
+		}
+	}
+
+	fixtureReleaseToManifests := func(release *repository.CreateApplicationVersion) *api.GetManifestsResponse {
+		return &api.GetManifestsResponse{
+			Release: &api.Release{
+				Version:        release.Version,
+				SourceCommitId: release.SourceCommitId,
+			},
+			Manifests: map[string]*api.Manifest{
+				"development": {
+					Environment: "development",
+					Content:     release.Manifests["development"],
+				},
+				"staging": {
+					Environment: "staging",
+					Content:     release.Manifests["staging"],
+				},
+			},
+		}
+	}
+
+	for _, tc := range []*testCase{
+		func() *testCase {
+			release := fixtureRelease(appName, 3)
+
+			return &testCase{
+				name: "happy path",
+				setup: append(fixtureSetupEnv(),
+					fixtureRelease(appNameOther, 1),
+					fixtureRelease(appNameOther, 2),
+					fixtureRelease(appName, 1),
+					fixtureRelease(appName, 2),
+					release,
+				),
+				req:  fixtureRequest(),
+				want: fixtureReleaseToManifests(release),
+			}
+		}(),
+		func() *testCase {
+			release := fixtureRelease(appName, 2)
+
+			return &testCase{
+				name: "request specific release",
+				setup: append(fixtureSetupEnv(),
+					fixtureRelease(appName, 1),
+					fixtureRelease(appNameOther, 1),
+					release,
+					fixtureRelease(appName, 3),
+					fixtureRelease(appNameOther, 2),
+				),
+				req:  fixtureRequest(func(req *api.GetManifestsRequest) { req.Release = "2" }),
+				want: fixtureReleaseToManifests(release),
+			}
+		}(),
+		{
+			name: "no release specified",
+			setup: append(fixtureSetupEnv(),
+				fixtureRelease(appName, 1),
+				fixtureRelease(appName, 2),
+				fixtureRelease(appName, 3),
+			),
+			req:     fixtureRequest(func(req *api.GetManifestsRequest) { req.Release = "" }),
+			wantErr: status.Error(codes.InvalidArgument, "invalid release number, expected uint or 'latest'"),
+		},
+		{
+			name: "no application specified",
+			setup: append(fixtureSetupEnv(),
+				fixtureRelease(appName, 1),
+				fixtureRelease(appName, 2),
+				fixtureRelease(appName, 3),
+			),
+			req:     fixtureRequest(func(req *api.GetManifestsRequest) { req.Application = "" }),
+			wantErr: status.Error(codes.InvalidArgument, "no application specified"),
+		},
+		{
+			name: "no releases for application",
+			setup: append(fixtureSetupEnv(),
+				fixtureRelease(appNameOther, 1),
+				fixtureRelease(appNameOther, 2),
+				fixtureRelease(appNameOther, 3),
+			),
+			req:     fixtureRequest(),
+			wantErr: status.Errorf(codes.NotFound, "no releases found for application %s", appName),
+		},
+	} {
+		tc := tc // TODO: Remove after switching to go v1.22
```

Comment:
no "TODO" comments without reference.
Please create a story to upgrade to 1.22 and share the story Id in the comment